File: 1704. Determine if String Halves Are Alike/Solution.java
import java.util.Set;

class Solution {
    public boolean halvesAreAlike(String s) {
        Set<Character> vowels = Set.of('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U');
        int half = s.length() / 2, cnt = 0;
        for (int i = 0; i < half; i++) {
            if (vowels.contains(s.charAt(i))) {
                cnt++;
            }
        }
        for (int i = half; i < s.length(); i++) {
            if (vowels.contains(s.charAt(i)) && --cnt < 0) {
                return false;
            }
        }
        return cnt == 0;
    }
}

File: 1704. Determine if String Halves Are Alike/Solution.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}
        n = len(s) // 2
        return sum(c in vowels for c in s[:n]) == sum(c in vowels for c in s[n:])
